pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -221,16 +452,61 @@ func (r *ReplicatedUpgradeReconciler) assignSubclustersToReplicaGroupACallback()
 	return annotatedAtLeastOnce, nil
 }
 
+// moveReplicaGroupBSubclusterToSandbox will move all subclusters attached to
+// replica group B into the sandbox.
+func (r *ReplicatedUpgradeReconciler) moveReplicaGroupBSubclusterToSandbox() (bool, error) {
+	oldImage, found := r.Manager.fetchOldImage()
+	if !found {
+		return false, errors.New("Could not find old image")
+	}
+
+	scNames := r.getSubclustersForReplicaGroup(vmeta.ReplicaGroupBValue)
+	if len(scNames) == 0 {
+		return false, errors.New("cound not find any subclusters for replica group B")
+	}
+
+	sandboxName, err := r.getNewSandboxName(preferredSandboxName)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate a unique sandbox name: %w", err)
+	}
+	sandbox := vapi.Sandbox{
+		Name:  sandboxName,
+		Image: oldImage,
+	}
+	for _, nm := range scNames {
+		sandbox.Subclusters = append(sandbox.Subclusters, vapi.SubclusterName{Name: nm})
+	}
+	r.VDB.Annotations[vmeta.ReplicatedUpgradeSandboxAnnotation] = sandboxName
+	r.VDB.Spec.Sandboxes = append(r.VDB.Spec.Sandboxes, sandbox)
+	return true, nil
+}
+
+// setImageInSandbox will set the new image in the sandbox to initiate an upgrade
+func (r *ReplicatedUpgradeReconciler) setImageInSandbox() (bool, error) {
+	sb := r.VDB.GetSandbox(r.sandboxName)
+	if sb == nil {
+		return false, fmt.Errorf("could not find sandbox %q", r.sandboxName)
+	}
+	sb.Image = r.VDB.Spec.Image
+	return true, nil
+}
+
 // countSubclustersForReplicaGroup is a helper to return the number of
 // subclusters assigned to the given replica group.
 func (r *ReplicatedUpgradeReconciler) countSubclustersForReplicaGroup(groupName string) int {
-	count := 0
+	scNames := r.getSubclustersForReplicaGroup(groupName)
+	return len(scNames)
+}
+
+// getSubclustersForReplicaGroup returns the names of the subclusters that are part of a replica group.
+func (r *ReplicatedUpgradeReconciler) getSubclustersForReplicaGroup(groupName string) []string {
+	scNames := []string{}
 	for i := range r.VDB.Spec.Subclusters {
```

Comment:
Should this be a function in v1?

Reply:
I'm not sure if we'll need this outside of replicated upgrade though. We can always move it later.